src/setfit/onnx.py

```diff
@@ -0,0 +1,281 @@
+import warnings
+from typing import Callable, Optional, Union
+
+import numpy as np
+import onnx
+import torch
+from sentence_transformers import SentenceTransformer, models
+from sklearn.linear_model import LogisticRegression
+from torch import nn
+from transformers.modeling_utils import PreTrainedModel
+
+
+def mean_pooling(token_embeddings: torch.Tensor, attention_mask: torch.Tensor) -> torch.Tensor:
+    """Perform attention aware mean pooling.
+
+    This method takes in embeddings of shape (batch, sequence, embedding_size) and performs average
+    pooling across the sequence dimension to yield embeddings of size (batch, embedding_size).
+
+    From:
+    https://huggingface.co/sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2#usage-huggingface-transformers
+
+    Args:
+        token_embeddings (`torch.Tensor`): The embeddings we wish to pool over of shape
+            (batch, sequence, embedding_size).  This will pool over the sequence to yield
+            (batch, embedding_size).
+        attention_mask (`torch.Tensor`): The binary attention mask across the embedings of shape
+
+    Returns:
+        (`torch.Tensor`) The mean pooled embeddings of size (batch, embedding_size).
+    """
+    input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
+    return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(input_mask_expanded.sum(1), min=1e-9)
+
+
+class OnnxSetFitModel(torch.nn.Module):
+    """A wrapper around SetFit model body, pooler, and model head which makes ONNX exporting easier.
+
+    This wrapper allows us to create a nn.Module with different levels of connectivity. We can set
+    model_body and pooler and have a Module which maps inputs to embeddings or we can set all three
+    and have a model which maps inputs to final predictions. This is useful because torch.onnx.export
+    will work with a nn.Module.
+
+    Attributes:
+        model_body (`PreTrainedModel`): The pretrained model body of a setfit model.
+        pooler (`nn.Module` or `Callable[[torch.Tensor], torch.Tensor]`): The callable function that can
```

Comment:
I think we can use `Union` here for these types
```suggestion
        pooler (`Union[nn.Module, Callable[[torch.Tensor], torch.Tensor]]`, *optional*, defaults to `None`): The callable function that can
```